### PR TITLE
Cisco ASA Add new action Create Address Object

### PR DESCRIPTION
**Pull request by r7-mgorny**
## Proposed Changes
Add new action Create Address Object

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/01_create_address_object_ipv4.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/02_create_address_object_ipv4_without_name.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/03_create_address_object_ipv6.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/04_create_address_object_ipv6_without_name.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/05_create_address_object_fqdn.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/06_create_address_object_fqdn_without_name.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/07_create_address_object_cidr.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/08_create_address_object_cidr_without_name.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/09_create_address_object_range.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/10_create_address_object_range_without_name.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/11_create_address_object_public_ip_with_skip.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nPrivate address found. Address object was RFC 1918 (private).",
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\nAn error occurred during plugin execution!\n\nPrivate address found. Address object was RFC 1918 (private).\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 307, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 428, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/actions/create_address_object/action.py\", line 26, in run\n    raise PluginException(\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nPrivate address found. Address object was RFC 1918 (private).\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/12_create_address_object_private_ip_with_skip_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nAddress Object not created because the host 111.113.111.111 was found in the whitelist as ['111.113.111.111']. If you would like to block this host, remove 111.113.111.111 from the whitelist and try again.",
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\nWhitelist matched\n111.113.111.111 was found in whitelist\nAn error occurred during plugin execution!\n\nAddress Object not created because the host 111.113.111.111 was found in the whitelist as ['111.113.111.111']. If you would like to block this host, remove 111.113.111.111 from the whitelist and try again.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 307, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 428, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/actions/create_address_object/action.py\", line 34, in run\n    raise PluginException(\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nAddress Object not created because the host 111.113.111.111 was found in the whitelist as ['111.113.111.111']. If you would like to block this host, remove 111.113.111.111 from the whitelist and try again.\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/13_create_address_object_ip_on_whitelist_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug run < tests/14_create_address_object_ip_with_whitelist.json
</summary>
</details>

### Validate

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: help.md[241]: https://example.com:443/api/objects/networkobjects/ASA_Demo_NObj_00
[*] Plugin successfully validated!

----
[*] Total time elapsed: 15801.987000000001ms

```

<summary>
icon-validate --all .
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug test < tests/01_create_address_object_ipv4.json
</summary>
</details>

Autogenerate with:
<details>

```
INFO[0000] Running command:  docker run --rm -i rapid7/cisco_asa:1.4.0 test < tests/01_create_address_object_ipv4_bad.json 
Connect: Connecting...
rapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object
Starting new HTTPS connection (1): asav-9-13-1.vuln.lax.rapid7.com:443
/usr/local/lib/python3.8/site-packages/urllib3-1.25.9-py3.8.egg/urllib3/connectionpool.py:979: InsecureRequestWarning: Unverified HTTPS request is being made to host 'asav-9-13-1.vuln.lax.rapid7.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
https://asav-9-13-1.vuln.lax.rapid7.com:443 "GET /api/monitoring/clock HTTP/1.1" 401 None
Connection test failed!
Invalid username or password provided. Verify your username and password are correct.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/connection/connection.py", line 28, in test
    clock = self.cisco_asa_api.get_clock()
  File "/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/util/api.py", line 83, in get_clock
    return self._call_api("GET", "monitoring/clock")
  File "/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/util/api.py", line 116, in _call_api
    raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD)
insightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!
Invalid username or password provided. Verify your username and password are correct.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py", line 307, in handle_step
    output = self.start_step(
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py", line 430, in start_step
    output = func()
  File "/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/connection/connection.py", line 35, in test
    raise ConnectionTestException(
insightconnect_plugin_runtime.exceptions.ConnectionTestException: Connection test failed!
Invalid username or password provided. Verify your username and password are correct.
{"body": {"log": "Connect: Connecting...\nrapid7/Cisco Adaptive Security Appliance:1.4.0. Step name: create_address_object\nConnection test failed!\n\nInvalid username or password provided. Verify your username and password are correct.\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/connection/connection.py\", line 28, in test\n    clock = self.cisco_asa_api.get_clock()\n  File \"/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/util/api.py\", line 83, in get_clock\n    return self._call_api(\"GET\", \"monitoring/clock\")\n  File \"/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/util/api.py\", line 116, in _call_api\n    raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD)\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nInvalid username or password provided. Verify your username and password are correct.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 307, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 430, in start_step\n    output = func()\n  File \"/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/icon_cisco_asa/connection/connection.py\", line 35, in test\n    raise ConnectionTestException(\ninsightconnect_plugin_runtime.exceptions.ConnectionTestException: Connection test failed!\n\nInvalid username or password provided. Verify your username and password are correct.\n", "status": "error", "meta": {}, "error": "Connection test failed!\n\nInvalid username or password provided. Verify your username and password are correct."}, "version": "v1", "type": "action_event"}
Traceback (most recent call last):
  File "/usr/local/bin/icon_cisco_asa", line 4, in <module>
    __import__('pkg_resources').run_script('cisco-asa-rapid7-plugin==1.4.0', 'icon_cisco_asa')
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 667, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1470, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/EGG-INFO/scripts/icon_cisco_asa", line 40, in <module>
  File "/usr/local/lib/python3.8/site-packages/cisco_asa_rapid7_plugin-1.4.0-py3.8.egg/EGG-INFO/scripts/icon_cisco_asa", line 36, in main
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/cli.py", line 228, in run
    args.func(args)
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/cli.py", line 153, in test_step
    return self.execute_step(is_test=True, is_debug=args.debug)
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/cli.py", line 147, in execute_step
    raise exception
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/cli.py", line 138, in execute_step
    output = self.plugin.handle_step(msg, is_test=is_test, is_debug=is_debug)
  File "/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-4.0.2-py3.8.egg/insightconnect_plugin_runtime/plugin.py", line 342, in handle_step
    raise LoggedException(ex, output)
insightconnect_plugin_runtime.exceptions.LoggedException: Connection test failed!
Invalid username or password provided. Verify your username and password are correct.

```

<summary>
docker run --rm -i rapid7/cisco_asa:1.4.0 --debug test < tests/01_create_address_object_ipv4_bad.json
</summary>
</details>

### CODE REVIEW

File: cisco_asa/icon_cisco_asa/actions/create_address_object/action.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    def run(self, params={}):
+        address = params.get(Input.ADDRESS)
+        name = params.get(Input.ADDRESS_OBJECT, address.replace(":", "-").replace("/", "-"))
+        whitelist = params.get(Input.WHITELIST)
+        skip_private = params.get(Input.SKIP_PRIVATE_ADDRESSES)
+
+        object_type = self.determine_address_type(address)
+        if skip_private and object_type != "fqdn" and self.check_if_private(address):
+            raise PluginException(
+                cause="Private address found.",
```
Comment (jschipp-r7) [inline cisco_asa/icon_cisco_asa/actions/create_address_object/action.py:27]:
```suggestion
                cause="Unable to create object for provided RFC 1918 (private) address.",
```

File: cisco_asa/icon_cisco_asa/actions/create_address_object/action.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                assistance="Address object was RFC 1918 (private)."
```
Comment (jschipp-r7) [inline cisco_asa/icon_cisco_asa/actions/create_address_object/action.py:28]:
```suggestion
                assistance="Provide a public address to create or disable Skip Private Addresses and try again."
```

File: cisco_asa/icon_cisco_asa/actions/create_address_object/action.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    @staticmethod
+    def check_if_private(address: str) -> bool:
+        if re.search('/', address):  # CIDR
+            return ip_network(address, False).is_private
+        elif re.search('-', address):  # IP Range
+            split_ = address.split("-")
+            if len(address.split("-")) != 2:  # If this isn't 2, I'm not sure what the input was
+                raise PluginException(
+                    cause="Wrong input",
```
Comment (jschipp-r7) [inline cisco_asa/icon_cisco_asa/actions/create_address_object/action.py:89]: Needs punctuation
```suggestion
                    cause="Improperly formatted input provided.",
```

File: cisco_asa/icon_cisco_asa/actions/create_address_object/action.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                    assistance="Range should have one and only one dash"
```
Comment (jschipp-r7) [inline cisco_asa/icon_cisco_asa/actions/create_address_object/action.py:90]:
```suggestion
                    assistance="Range should have one and only one dash."
```

File: cisco_asa/plugin.spec.yaml [hunk 5]
```diff
@@ -189,3 +189,37 @@ actions:
         description: Success if address add to group
         type: boolean
         required: true
+  create_address_object:
+    title: Create Address Object
+    description: Create address object by the object IP address
+    input:
+      address:
+        title: Address
+        description: IP, CIDR, or domain name to assign to the address object
```
Comment (jschipp-r7) [inline cisco_asa/plugin.spec.yaml:198]: This says domain and CIDR is supported when the action description on line 194 says "IP address"

We need to update one of these lines to be correct and match the other.

File: cisco_asa/plugin.spec.yaml [hunk 6]
```diff
@@ -189,3 +189,37 @@ actions:
...
+        type: string
+        required: true
+        example: 198.51.100.100
+      address_object:
+        title: Address Object
+        description: Name of the address object, defaults to the value address in the address field if no name is given
+        type: string
+        required: false
+        example: InsightConnect Address Object
```
Comment (jschipp-r7) [inline cisco_asa/plugin.spec.yaml:207]:
```suggestion
        example: MaliciousHost
```

File: cisco_asa/plugin.spec.yaml [hunk 7]
```diff
@@ -189,3 +189,37 @@ actions:
...
+    input:
+      address:
+        title: Address
+        description: IP, CIDR, or domain name to assign to the address object
+        type: string
+        required: true
+        example: 198.51.100.100
+      address_object:
+        title: Address Object
+        description: Name of the address object, defaults to the value address in the address field if no name is given
```
Comment (jschipp-r7) [inline cisco_asa/plugin.spec.yaml:204]:
```suggestion
        description: Name of the address object, defaults to the value in the address field if no name is given
```

File: cisco_asa/icon_cisco_asa/actions/create_address_object/action.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        if re.search(r'[a-zA-Z]', address):
+            return "IPv4FQDN"
+        if '-' in address:
+            return "IPv4Range"
+
+        return "IPv4Address"
+
+    def match_whitelist(self, address: str, whitelist: list, object_type: str) -> str:
+        if address in whitelist:
+            self.logger.info(f"Whitelist matched\n{address} was found in whitelist")
```
Comment (jschipp-r7) [inline cisco_asa/icon_cisco_asa/actions/create_address_object/action.py:61]:
```suggestion
            self.logger.info(f"Whitelist matched\n{address} was found in whitelist.")
```

File: cisco_asa/icon_cisco_asa/actions/create_address_object/action.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        # if 1.1.1.1/32 - remove /32
+        trimmed_address = re.sub(r"/32$", "", address)
+
+        # IP is in CIDR - Give the user a log message
+        for item in whitelist:
+            if self.determine_address_type(item) == "IPv4Network":
+                net = ip_network(item, False)  # False means ignore the masked bits, otherwise they need to be 0
+                ip = ip_address(trimmed_address)
+                if ip in net:
+                    self.logger.info(f"Whitelist matched\nIP {address} was found in {item}")
```
Comment (jschipp-r7) [inline cisco_asa/icon_cisco_asa/actions/create_address_object/action.py:76]:
```suggestion
                    self.logger.info(f"Whitelist matched\nIP {address} was found in {item}.")
```

File: cisco_asa/help.md [hunk 10]
```diff
@@ -54,6 +55,46 @@ Example input:
...
+|address_object|string|None|False|Name of the address object, defaults to the value address in the address field if no name is given|None|InsightConnect Address Object|
+|skip_private_addresses|boolean|None|False|If set to true, any addresses that are defined in the RFC1918 space will not be blocked. e.g. 10/8, 172.16/12, 192.168/16|None|False|
+|whitelist|[]string|None|False|This list contains a set of hosts that should not be blocked. This can include IPs, CIDRs, and domains|None|["198.51.100.100"]|
+
+Example input:
+
+```
+{
+  "address": "198.51.100.100",
+  "address_object": "InsightConnect Address Object",
```
Comment (jschipp-r7) [inline cisco_asa/help.md:76]:
```suggestion
  "address_object": "MaliciousHost",
```

File: cisco_asa/plugin.spec.yaml (hunk 5 above)
Comment (r7-mgorny) [reply cisco_asa/plugin.spec.yaml]: @jschipp-r7  I changed this description in both places. Action allow to create address object with ipv4, ipv6, fqdn, cidr and range.

**Pull request merged**